src/OrchardCore.Modules/OrchardCore.Forms/Drivers/FormPartDisplayDriver.cs

```diff
@@ -1,23 +1,51 @@
+using System;
```

Comment:
The token is used in the context of a connection dedicated to a given user and not intended to be exchanged, but yes can be still stolen, even more if its lifetime is too long.

What we allow here is to corellate a form with a WF, in that case you don't have to generate a token from the UI, it is auto generated dynamically on each submission.

@deanmarcussen did a secret type to correlate a form to a WF, here we do the same but without using a secret, a given form can be still correlated to a WF from the UI and in that case a token is still created dynamically on each submission.

So the user has the choice to generate a token as before or to select a WF, in that case the "static" token is no longer rendered, a new one will be auto generated on each submission.

Reply:
In one case we dynamically create a token on the form submission. In the other cases we need to provide to people an URL containing a token, so the token needs to be already generated and then it can't be dynamically generated.

Reply:
My issue is with `HttpWorkflowController.Invoke()`: There, we use a persisted token to invoke a workflow via an HTTP request. So, the token, which is a secret-like value, IMO should be saved to the DB like a password, as a secret. However, it's currently using `ISecurityTokenService` which in turn uses `IDataProtectionProvider`.

So, this value is not portable, and thus e.g. such workflows can't be exported in a way that they'll retain their URLs (and thus e.g. forms will break if you move them to a different environment, or if you initialize a site from a recipe).

Reply:
If it is intended to be used with a form, you now can correlate the form to a workflow type, in that case in the UI the URL generation is no longer rendered and the token will be created dynamically.

But if you need to share a protected URL including a token as a link how can you do that.

Reply:
So just to confirm: if you use this with a form, then you'll be able to export-import or the form and the workflow, and it'll keep working, no need to regenerate the URL token, right? But it'll be needed to be regenerated if you use the Http Request event to call it from somewhere else (like an external webhook call).

Reply:
Exactly but better said, sorry for my english.

Reply:
I see, thanks. How complex would it be to not to have to regenerate the token otherwise either?

Reply:
For example if you have to 2 servers with a stateless configuration and intended to be in sync by sharing the same database, Redis instance and other distributed services, no problem but in that case you don't need to export/import anything.

Otherwise, the "problem" is around the data protection which would not use the same keys, but you can try to only use our distributed data protection feature, but this would imply for a given tenant (same name) to use the same Redis instance.

Reply:
I'm not really thinking of a muti-server setup (that you wouldn't want to handle with export-import as you say) but rather actual export and import, when you want to move data between completely independent instances (like staging to prod, or from your local dev env to prod, or between dev environments of team members).

But thinking of it, in these cases, most possibly the host part of the URL (i.e. domain) would also change, so changing a configured URL for an Http Request Task-started workflow (i.e. a webhook) that external services call would be necessary anyway.

The only exception is the local development scenario: Consider your team having such a webhook to start a workflow when something in an external DB is updated. So, in that DB you implement something to call the webhook. Now, you can run it on your local machine to test, but if your teammate reinitializes their local OC dev instance from a shared recipe (which is the usual approach), then to test it too (like during code review), they'll need to regenerate the token and change it in the DB update event too. Not a huge deal though.

Reply:
@Piedone 

Yes I understand, hmm maybe you were right, maybe this is `ISecurityTokenService` that would need to be overriden, or another interface `ISecretTokenService` to be used as needed.

So that in the definition of an Http Request Task we could select Secret(s), not to hide the generated URL but to be used to generate the url token and that will be used for decryption.

Or I think settings to define Secret(s) (maybe only one for encryption, not for signing) to be used by `ISecretTokenService`, so that we don't need to select/create secrets per Http Request Task.

Need to focus on it again.

Reply:
Let me know, then.